### PR TITLE
Avoid computing end condition on each iteration

### PR DESCRIPTION
**Pull request by sdenel**

### CODE REVIEW

File: src/vs/base/common/arrays.ts [hunk 1]
```diff
@@ -171,7 +171,7 @@ export function distinct<T>(array: T[], keyFn?: (t: T) => string): T[] {
 }
 
 export function first<T>(array: T[], fn: (item: T) => boolean, notFoundValue: T = null): T {
-	for (let i = 0; i < array.length; i++) {
+	for (let i = 0, I = array.length; i < I; i++) {
 		const element = array[i];
```
Comment (jrieken) [inline src/vs/base/common/arrays.ts]: we don't do upper-case local

File: src/vs/base/common/arrays.ts [hunk 2]
```diff
@@ -171,7 +171,7 @@ export function distinct<T>(array: T[], keyFn?: (t: T) => string): T[] {
 }
 
 export function first<T>(array: T[], fn: (item: T) => boolean, notFoundValue: T = null): T {
-	for (let i = 0; i < array.length; i++) {
+	for (let i = 0, iMax = array.length; i < iMax; i++) {
```
Comment (jrieken) [inline src/vs/base/common/arrays.ts]: sorry, actually we should use a ES6 for-of loop

Comment (sdenel) [inline src/vs/base/common/arrays.ts]: You are absolutely right, I didn't know that construct. See my new patch.

**Comment by sdenel** [comment]
Sorry for the noise, I will do changes in my own branch and come back later.

**Comment by jrieken** [comment]
ok

**Pull request closed without merging**